### DISCUSSION TITLE
mogrify -profile -format FILE doesn't apply colorprofile

### DISCUSSION THREAD

**Discussion by aakkeemm** [ImageMagick/ImageMagick#7823] [Help]
### ImageMagick version

7.1.1-41 Q16-HDRI x64

### Operating system, version and so on

Windows 10 HOME

### Description

Hi,

i wonder why
`mogrify -profile ".\sRGB Color Space Profile.icm" -format JPG testfile.tif`
or
`mogrify -colorspace sRGB -format JPG testfile.tif`
do write out a testfile.jpg file but don't apply the sRGB Profile.
testfile.tif has a colorprofile (eciRGB v2) embedded.

Same with 7.1.1-41 Q16-HDRI x86_64 on macOS 12.7.6
`mogrify -profile ./sRGB\ Color\ Space\ Profile.icm -format jpg ./testfile.tif`

**Comment by snibgo** [comment]
I suggest using "magick mogrify" instead of just "mogrify". But that doesn't solve the problem.

I confirm the problem. "-debug all" reveals that IM is trying to read the ICC (or ICM) file as an image. That fails and is reported by "-debug all", but not reported otherwise. This seems to be a bug.

The workaround for single files is not to use "magick mogrify", but instead use just "magick":
```
magick testfile.tif -profile ".\sRGB Color Space Profile.icm" testfile.jpg
```
